go.go

```diff
@@ -99,12 +99,26 @@ func Go(a, b []byte) (changes []*Change, err error) {
 			ds := x.diffstat(ai, bi)
 			changes = append(changes, &Change{Name: s, DelLines: ds.del, InsLines: ds.ins})
 		} else {
-			changes = append(changes, &Change{Name: s, Inserted: true, InsLines: len(bytes.Split(x.bBytes(bi), newline))})
+			var ln int
+			if len(ins) == len(del) {
+				ds := x.diffstat(bi, bi)
```

Comment:
This doesn't smell right. `bi` is an index into `b`, not `a`. Can you write a comment clarifying why this is correct?